cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/AuthenticationStrategyProvider.java

```diff
@@ -38,15 +37,13 @@ public AuthenticationStrategyProvider(String namespaceId, SecureStore secureStor
    * Returns an {@link AuthenticationStrategy} for the given Git repository provider and auth type.
    *
    * @return an instance of {@link  AuthenticationStrategy}.
-   * @throws AuthenticationStrategyNotFoundException when the corresponding {@link
-   *     AuthenticationStrategy} is not found.
+   * @throws AuthenticationStrategyNotFoundException when the corresponding
+   *     {@link AuthenticationStrategy} is not found.
    */
   AuthenticationStrategy get(RepositoryConfig repoConfig) throws
       AuthenticationStrategyNotFoundException {
-    if (repoConfig.getProvider() == Provider.GITHUB) {
-      if (repoConfig.getAuth().getType() == AuthType.PAT) {
-        return new GitPatAuthenticationStrategy(secureStore, repoConfig, namespaceId);
-      }
+    if (repoConfig.getAuth().getType() == AuthType.PAT) {
```

Comment:
since we only support `PAT` authentication strategy for `github/gitlab`.

Do we need an additional check here?

I know currently there are only two providers but for future do we need it?

Reply:
Currently I don't think we necessarily need this as AuthType enum only have one value.
But in future we might have requirement for oauth2 and also we most probably add new authtype for bitbucket as it is not strictly PAT. Hence better to keep this here for now. 
I will remove this check if we do not need to add a new authtype for bitbucket